scaper/core.py

```diff
@@ -1307,14 +1309,30 @@ def _instantiate_event(self, event, isbackground=False,
                 if source_time + event_duration > source_duration:
                     source_time = source_duration - event_duration
                     warn = True
+                    tuple_still_invalid = True
 
             if warn and not disable_instantiation_warnings:
-                warnings.warn(
-                    '{:s} source time tuple went out of bounds '
-                    'source duration ({:.2f}) and event duration ({:.2f}), modified source time '
-                    'tuple to stay within bounds'.format(
-                        label, source_duration, event_duration),
-                    ScaperWarning)
+                old_source_time = ', '.join(map(str, event.source_time))
+                new_source_time = ', '.join(map(str, modified_source_time))
+                if not tuple_still_invalid:
+                    warnings.warn(
+                        "{:s} source time tuple ({:s}) could not be satisfied given "
+                        "source duration ({:.2f}) and event duration ({:.2f}), "
+                        "source time tuple changed to ({:s})".format(
+                            label, old_source_time, source_duration, 
+                            event_duration, new_source_time), 
+                        ScaperWarning)
+                else:
+                    warnings.warn(
+                        "{:s} source time tuple ({:s}) could not be satisfied given "
+                        "source duration ({:.2f}) and event duration ({:.2f}), "
+                        "source time tuple changed to ({:s}) but still did not satisfy "
+                        "the condition, likely due to using 'normal' distribution with "
+                        "bounds too close to the start or end of the audio file".format(
+                            label, old_source_time, source_duration, 
+                            event_duration, new_source_time),
+                        ScaperWarning)
+
         else:
             source_time = 0.0
 
```

Comment:
I'm open to editing the second warning a bit. Let me know of better text and I'll toss it in there, unless you think this is alright!